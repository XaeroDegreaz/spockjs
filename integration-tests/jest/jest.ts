import { resolve } from 'path';

import { modulePath, runWithTypescriptJit } from '../utils';

const jestCli = resolve(modulePath('jest-cli'), 'bin', 'jest');

export const runJest = (cwd: string) => {
  const { status, stdout } = runWithTypescriptJit(
    [jestCli, '--no-cache', '--noStackTrace', '--runInBand', '--json'],
    { cwd },
  );

  return { status, result: JSON.parse(stdout.toString()) };
};
