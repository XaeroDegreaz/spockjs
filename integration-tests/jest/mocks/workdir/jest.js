test('stubbing', () => {
  const s = jest.fn();
  stub: s() >> 42;
  expect(s()).toBe(42);
});

test('mocking', () => {
  const mock = jest.fn();
  mock: 2 * mock();

  mock();
  mock();

  verify: mock;
});

test('failing verification', () => {
  const mock = jest.fn();
  mock: 2 * mock();

  mock();

  verify: mock;
});
