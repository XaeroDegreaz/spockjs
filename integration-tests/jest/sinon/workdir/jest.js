import * as sinon from 'sinon';

test('stubbing', () => {
  const s = sinon.stub();
  stub: s() >> 42;
  expect(s()).toBe(42);
});

test('mocking', () => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();
  obj.method();

  verify: mock;
});

test('failing verification', () => {
  const obj = { method: () => {} };

  const mock = sinon.mock(obj);
  mock: 2 * mock.method();

  obj.method();

  verify: mock;
});
