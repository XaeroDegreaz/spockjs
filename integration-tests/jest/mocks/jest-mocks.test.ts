import { resolve } from 'path';

import { runJest } from '../jest';

// mark implicit dependencies for jest
() => require('./workdir/jest.js') && require('./workdir/package.json');
() => require('@spockjs/preset-runner-jest');
() => require('@spockjs/preset-jest-mocks');

const cwd = resolve(__dirname, 'workdir');

test('interaction blocks work', () => {
  const {
    status,
    result: {
      numPassedTests,
      numFailedTests,
      testResults: [{ message }],
    },
  } = runJest(cwd);

  expect: {
    status === 1;
    numPassedTests === 2;
    numFailedTests === 1;
  }
  expect(message).toMatchSnapshot();
});
