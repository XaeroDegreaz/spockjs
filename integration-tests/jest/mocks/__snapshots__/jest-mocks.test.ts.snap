// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interaction blocks work 1`] = `
"  ‚óè failing verification

    Expected 2 call(s) to mock jest.fn() with arguments Array [], but received 1 such call(s).
"
`;
